GalilSup/src/GalilController.cpp

```diff
@@ -2874,15 +2879,18 @@ asynStatus GalilController::writeReadController(const char *caller)
   	try	{
 		//If connected, then do the command
 		if (gco_ != NULL)
-		       	{
+		{
 		 	//send the command, and get the response
 			strncpy(resp_, gco_->command(strcmd).c_str(), sizeof(resp_));
 			resp_[sizeof(resp_)-1] = '\0';
 			//No exception = success
 			done = true;
-		 	consecutive_timeouts_ = 0;
-			status = asynSuccess;
+			// Only track consecutive timeouts for same messages as in catch
+			if (strncmp(cmd_, "BP",2)==0 || strncmp(cmd_, "MT",2) == 0) {
+				consecutive_read_timeouts_ = 0;
 			}
```

Comment:
This will only reset consecutive_read_timeouts_ on a subsequent successful BP or MT command - is that intended? This logic would appear to be different from the original code, and a code comment suggests BP always causes a timeout so that part of the branch would never get executed.